### PR TITLE
add roles and descriptions to carousel component

### PR DESCRIPTION
**Pull request by uraaaa24**
Closes https://github.com/yamada-ui/yamada-ui/issues/1728

## Description

add a11y to `carousel`

https://www.w3.org/WAI/ARIA/apg/patterns/carousel/examples/carousel-1-prev-next/
https://www.w3.org/WAI/ARIA/apg/patterns/carousel/

## Is this a breaking change (Yes/No):

No

### CODE REVIEW

**Review by kaehehehe — APPROVED** [review]

**Review by hirotomoyamada — CHANGES_REQUESTED** [review]
Set `aria-label`, `aria-live` and `aria-controls`.

**Comment by hirotomoyamada** [comment]
@uraaaa24 

Please check quolity test and lint.

**Comment by uraaaa24** [comment]
@hirotomoyamada 
Sorry for the late confirmation.
I've fixed it, please check back.

**Review by hirotomoyamada — CHANGES_REQUESTED** [review]
The `id` must be unique.This will be violated if there are multiple `Carousels` on the screen.

File: packages/components/carousel/src/use-carousel.ts [hunk 1]
```diff
@@ -494,7 +506,9 @@ export const useCarouselIndicators = () => {
       const isSelected = index === selectedIndex
 
       return {
```
Comment (hirotomoyamada) [inline packages/components/carousel/src/use-carousel.ts:508]: Is `aria-controls` not required?

File: packages/components/carousel/src/use-carousel.ts [hunk 2]
```diff
@@ -366,7 +367,12 @@ export const useCarousel = ({
     (props = {}, ref = null) => ({
       ...containerProps,
       ...props,
+      id: "carousel-items",
```
Comment (hirotomoyamada) [inline packages/components/carousel/src/use-carousel.ts:370]: This should be set to `getSlidesProps`.

File: packages/components/carousel/src/use-carousel.ts [hunk 3]
```diff
@@ -366,7 +367,12 @@ export const useCarousel = ({
...
       ref,
+      "aria-label": "Highlighted sections",
+      "aria-live": autoplay ? "off" : "polite",
```
Comment (hirotomoyamada) [inline packages/components/carousel/src/use-carousel.ts:373]: This should be set to `getSlidesProps`.

File: packages/components/carousel/src/use-carousel.ts (hunk 1 above)
Comment (uraaaa24) [reply packages/components/carousel/src/use-carousel.ts:508]: I had `aria-controls` set by `getControlProps`.

```ts
  const getControlProps: PropGetter<"button"> = useCallback(
    (props = {}, ref = null) => ({
      ...props,
      ref,
      "aria-controls": "carousel-items",
      disabled,
      onClick: handlerAll(props.onClick, onClick),
    }),
    [disabled, onClick],
  )
```

Does this mean that this is also a mistake?

Comment (uraaaa24) [reply packages/components/carousel/src/use-carousel.ts:508]: Refer to the end of the table below.

https://www.w3.org/WAI/ARIA/apg/patterns/carousel/examples/carousel-1-prev-next/#rps_label

Comment (hirotomoyamada) [reply packages/components/carousel/src/use-carousel.ts:508]: No, I think we need to set it on the control button as well and have `aria-controls` on the indicator as well. I thought so.

Comment (uraaaa24) [reply packages/components/carousel/src/use-carousel.ts:508]: Thanks for letting me know! I will respond that way!

**Comment by uraaaa24** [comment]
@hirotomoyamada 
I have finished correcting the points you pointed out!
Please check it!

**Review by hirotomoyamada — DISMISSED** [review]

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**